### PR TITLE
Add basic cast support

### PR DESCRIPTION
**Pull request by krocard**
This introduces Basic cast support for Flutter. The API is align as much as possible with the [React Native SDK](https://github.com/bitmovin/bitmovin-player-react-native/pull/262).

## New API
 - `BitmovinCastManager`
 - `player` cast methods (`castVideo`, `castStop`, `isCasting`...)

### Not yet supported
 - Cast events

## Issues encountered
 - The Cast SDK expects an AppCompat `Activity` (`FragmentActivity`), which the recommended `FlutterActivity` is not.
     - `MainActivity` now extends `FlutterFragmentActivity`
 - The Cast SDK expects an AppCompat style
     - The `Theme.AppCompat.NoActionBar` is now used instead of `LaunchApp`
 - `com.bitmovin.player.casting.ExpandedControllerActivity` was not accessible in the sample app
    - Made the player an `api` dependency

### `BitmovinPlayerManager`
Flutter's `Player` is a plain object with an async initialization. Aka, the Dart Player object is created immediately, but it's resources (native player, channels...) are created in the background. The client doesn't know when it is done [^1].
This was chosen for convenience as Dart has no async constructor.

The drawback of this approach is that the client code doesn't control the initialization sequence of the object explitly.

Given that the `BitmovinPlayerManager` must be created before the `Player`, the same patter is not applicable. Instead the `BitmovinPlayerManager` singleton is accessible through an async static `create` factory. This ensure that the client can initialize it reliably before creating a `Player`.

Nevertheless as the player creation is not async, mixing the async and sync code is not very convenient and leads to the cast use case needing a more complex setup than the non cast use-case. This is reflected in the complexity of the cast sample included in this PR.

One possible solution to avoid this would be to implicitly create the cast manager on the platform side if the player config has cast enabled set to true. Nevertheless this is no inlined with our other SDKs, which need explicit initialization of the `BitmovinCastManager`. As such this alternative has not been implemented, but **feedback on this proposal is welcomed**.

[^1]: The async part is implicitly `await`ed when an async method of the player is called. This is an implementation detail that should not be relied on by the client.

### `BitmovinPlayerManager.updateContext`
`updateContext` is supposed to be called for any new activity. Nevertheless the Flutter native plugin already gets a callback on Activity attach and detach. This PR has `updateContext` called automatically if the `BitmovinPlayerManager` is initialized.

As such there should be no need to call `updateContext` from the Dart side. It is still exposed for alignment with React Native and because I didn't full understand the Flutter plugin Activity lifecycle when I started this PR.

I would recommend removing `updateContext` from the Dart API, and **would welcome feedback on this proposal.**

### CODE REVIEW

**Comment by krocard** [comment]
@rolandkakonyi @strangesource Given that you did the React Native implementation, feel free to take a look at the Flutter one.
Note that we will wait for @hawk23's review to merge.

File: lib/src/api/casting/casting_config.dart [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  Map<String, dynamic> toJson() => _$RemoteControlConfigToJson(this);
+
+  /// A URL to a CSS file the receiver app loads to style the receiver app.
+  /// Default value is `null`, indicating that the default CSS of the receiver
+  /// app will be used.
+  final String? receiverStylesheetUrl;
+
+  /// A Map containing custom configuration values that are sent to the remote
```
Comment (strangesource) [inline lib/src/api/casting/casting_config.dart:30]:
```suggestion
  /// A map containing custom configuration values that are sent to the remote
```

**Review by strangesource — COMMENTED** [review]
Disclaimer: This review is very superficial as I haven't touched the Flutter SDK yet.

If casting is initialized for one example I think it is available in all samples. In order to avoid this somewhat confusing behaviour we should disable casting in the existing samples.

**Review by rolandkakonyi — COMMENTED** [review]
I literally have no idea what is going on. I never touched Flutter.
The Android part looks good as far as I can tell.

**Review by matamegger — COMMENTED** [review]
From my side nothing to add to the other reviews. I only shallowly looked over the code, this looks fine, but I basically have no experience with flutter (yet)

File: lib/src/casting/casting_config.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'casting_config.g.dart';
+
+/// Arguments to send a message
+@JsonSerializable(explicitToJson: true)
+class BitmovinCastManagerSendMessage extends Equatable {
```
Comment (hawk23) [inline lib/src/casting/casting_config.dart:8]: Nit: How about we go for a crisper name, e.g. `CustomCastMessage` or just `CastMessage`

File: lib/src/casting_manager.dart
Comment (hawk23) [inline lib/src/casting_manager.dart:1]: Should we move the cast manager to `src/casting`? I feel like that would be the proper place

File: lib/src/methods.dart [hunk 3]
```diff
@@ -21,6 +21,10 @@ class Methods {
   static const String setSubtitle = 'setSubtitle';
   static const String getSubtitle = 'getSubtitle';
   static const String removeSubtitle = 'removeSubtitle';
+  static const String isCastAvailable = 'isCastAvailable';
+  static const String isCasting = 'isCasting';
+  static const String castVideo = 'castVideo';
+  static const String castStop = 'castSTop';
```
Comment (hawk23) [inline lib/src/methods.dart:27]:
```suggestion
  static const String castStop = 'castStop';
```

File: lib/src/casting/casting_config.dart [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'casting_config.g.dart';
+
+/// Arguments to send a message
```
Comment (hawk23) [inline lib/src/casting/casting_config.dart:6]:
```suggestion
/// Arguments to send a message to the cast receiver.
```

File: lib/src/casting/casting_config.dart [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    this.messageNamespace,
+  });
+
+  factory BitmovinCastManagerSendMessage.fromJson(Map<String, dynamic> json) {
+    return _$BitmovinCastManagerSendMessageFromJson(json);
+  }
+
+  Map<String, dynamic> toJson() => _$BitmovinCastManagerSendMessageToJson(this);
+
+  /// The message to send
```
Comment (hawk23) [inline lib/src/casting/casting_config.dart:20]:
```suggestion
  /// The message to send.
```

File: lib/src/casting/casting_config.dart [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  final String message;
+  /// A custom message namespace to be used for communication between sender and
+  /// receiver.
+  /// Using `null` value will result in using the default message namespace
```
Comment (hawk23) [inline lib/src/casting/casting_config.dart:24]:
```suggestion
  /// Using `null` value will result in using the default message namespace.
```

File: lib/src/api/casting/casting_config.dart [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  /// A map containing custom configuration values that are sent to the remote
+  /// control receiver.
+  /// Default value is an empty map.
+  final Map<String, String> customReceiverConfig;
+
+  /// Whether casting is enabled.
+  /// Default value is `true`.
+  ///
+  /// Has no effect if the `BitmovinCastManager` is not initialized before the
+  /// `Player` is created with this configuration.
```
Comment (hawk23) [inline lib/src/api/casting/casting_config.dart:39]: Symbol references should be done using square brackets. Counts for all occurrences in this PR

File: lib/src/api/casting/casting_config.dart [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+  final bool isCastEnabled;
+
+  /// Indicates whether cookies and credentials will be sent along manifest
+  /// requests on the cast receiver
```
Comment (hawk23) [inline lib/src/api/casting/casting_config.dart:43]:
```suggestion
  /// requests on the cast receiver.
```

File: lib/src/api/casting/casting_config.dart [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+  /// Default value is `false`.
+  final bool sendManifestRequestsWithCredentials;
+
+  /// Indicates whether cookies and credentials will be sent along segment
+  /// requests on the cast receiver
```
Comment (hawk23) [inline lib/src/api/casting/casting_config.dart:48]:
```suggestion
  /// requests on the cast receiver.
```

File: lib/src/api/casting/casting_config.dart [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+  /// Default value is `false`.
+  final bool sendSegmentRequestsWithCredentials;
+
+  /// Indicates whether cookies and credentials will be sent along DRM licence
+  /// requests on the cast receiver
```
Comment (hawk23) [inline lib/src/api/casting/casting_config.dart:53]:
```suggestion
  /// requests on the cast receiver.
```

File: lib/src/api/casting/casting_config.dart [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+  const BitmovinCastManagerOptions({this.applicationId, this.messageNamespace});
+
+  factory BitmovinCastManagerOptions.fromJson(Map<String, dynamic> json) {
+    return _$BitmovinCastManagerOptionsFromJson(json);
+  }
+
+  Map<String, dynamic> toJson() => _$BitmovinCastManagerOptionsToJson(this);
+
+  /// ID of receiver application.
+  /// Using `null` value will result in using the default application ID
```
Comment (hawk23) [inline lib/src/api/casting/casting_config.dart:80]:
```suggestion
  /// Using `null` value will result in using the default application ID.
```

File: lib/src/api/casting/casting_config.dart [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+  final String? applicationId;
+  /// A custom message namespace to be used for communication between sender and
+  /// receiver.
+  /// Using `null` value will result in using the default message namespace
```
Comment (hawk23) [inline lib/src/api/casting/casting_config.dart:84]:
```suggestion
  /// Using `null` value will result in using the default message namespace.
```

File: lib/src/api/player/player_api.dart [hunk 13]
```diff
@@ -86,4 +86,19 @@ abstract class PlayerApi {
 
   /// Provides access to Analytics related functionality.
   AnalyticsApi get analytics;
+
+  /// Whether casting to a cast-compatible remote device is available.
+  /// `CastAvailableEvent` signals when casting becomes available.
```
Comment (hawk23) [inline lib/src/api/player/player_api.dart:91]: Let's not forget to use square bracket notation once the events exits 👀

File: lib/src/api/player/player_api.dart [hunk 14]
```diff
@@ -86,4 +86,19 @@ abstract class PlayerApi {
...
+
+  /// Whether video is currently being casted to a remote device and not played
+  /// locally.
+  Future<bool> get isCasting;
+
+  /// Initiates casting the current video to a cast-compatible remote device.
+  /// The user has to choose to which device it should be sent.
+  Future<void> castVideo();
+
+  /// Stops casting the current video. Has no effect if `isCasting` is false.
```
Comment (hawk23) [inline lib/src/api/player/player_api.dart:102]: Nit: It actually has an effect when `isCasting` is `false`. It opens the cast dialog, as if `castVideo()` has been called. Maybe we should adapt the doc in this regard.

File: lib/src/api/player/player_config.dart [hunk 15]
```diff
@@ -49,6 +50,9 @@ class PlayerConfig extends Equatable {
   @JsonKey(name: 'analyticsConfig')
   final AnalyticsConfig? analyticsConfig;
 
+  /// Configures remote playback functionality
```
Comment (hawk23) [inline lib/src/api/player/player_config.dart:53]:
```suggestion
  /// Configures remote playback functionality.
```

File: lib/src/player.dart [hunk 16]
```diff
@@ -285,6 +277,19 @@ class Player with PlayerEventHandler implements PlayerApi {
...
+  Future<bool> get isCastAvailable => _invokeMethod(Methods.isCastAvailable);
+
+  @override
+  Future<bool> get isCasting => _invokeMethod(Methods.isCasting);
+
+  @override
+  Future<void> castVideo() async => _invokeMethod<void>(Methods.castVideo);
+
+  @override
+  Future<void> castStop() => _invokeMethod<void>(Methods.castStop);
```
Comment (hawk23) [inline lib/src/player.dart:291]: Let's declare all of them async

**Review by hawk23 — COMMENTED** [review]
Wow, big feature! 🎉 

Some general things I noticed:
- There are some `*.g.dart` files missing in this PR
- I saw the Bitmovin UI showing "A general error occurred" a couple of times during manual testing. I do not have steps to reproduce though. Once I tried to reproduce it manually it never happened again. Maybe the stack trace from below helps

```
E/MediaCodecVideoRenderer(22389): Video codec error
E/MediaCodecVideoRenderer(22389):   java.lang.IllegalStateException
E/MediaCodecVideoRenderer(22389):       at android.media.MediaCodec.native_dequeueInputBuffer(Native Method)
E/MediaCodecVideoRenderer(22389):       at android.media.MediaCodec.dequeueInputBuffer(MediaCodec.java:2726)
E/MediaCodecVideoRenderer(22389):       at com.bitmovin.android.exoplayer2.mediacodec.SynchronousMediaCodecAdapter.dequeueInputBufferIndex(SynchronousMediaCodecAdapter.java:99)
E/MediaCodecVideoRenderer(22389):       at com.bitmovin.android.exoplayer2.mediacodec.MediaCodecRenderer.feedInputBuffer(MediaCodecRenderer.java:1180)
E/MediaCodecVideoRenderer(22389):       at com.bitmovin.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:775)
E/MediaCodecVideoRenderer(22389):       at com.bitmovin.android.exoplayer2.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:1017)
E/MediaCodecVideoRenderer(22389):       at com.bitmovin.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:512)
E/MediaCodecVideoRenderer(22389):       at android.os.Handler.dispatchMessage(Handler.java:106)
E/MediaCodecVideoRenderer(22389):       at android.os.Looper.loop(Looper.java:219)
E/MediaCodecVideoRenderer(22389):       at android.os.HandlerThread.run(HandlerThread.java:67)
E/ExoPlayerImplInternal(22389): Playback error
E/ExoPlayerImplInternal(22389):   com.bitmovin.android.exoplayer2.ExoPlaybackException: MediaCodecVideoRenderer error, index=0, format=Format(1080_4800000, null, null, video/avc, avc1.4D4032, 4800000, null, [1920, 1080, 25.0], [-1, -1]), format_supported=YES
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:588)
E/ExoPlayerImplInternal(22389):       at android.os.Handler.dispatchMessage(Handler.java:106)
E/ExoPlayerImplInternal(22389):       at android.os.Looper.loop(Looper.java:219)
E/ExoPlayerImplInternal(22389):       at android.os.HandlerThread.run(HandlerThread.java:67)
E/ExoPlayerImplInternal(22389):   Caused by: com.bitmovin.android.exoplayer2.video.MediaCodecVideoDecoderException: Decoder failed: OMX.hisi.video.decoder.avc
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.video.MediaCodecVideoRenderer.createDecoderException(MediaCodecVideoRenderer.java:1639)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:794)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:1017)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:512)
E/ExoPlayerImplInternal(22389):       ... 3 more
E/ExoPlayerImplInternal(22389):   Caused by: java.lang.IllegalStateException
E/ExoPlayerImplInternal(22389):       at android.media.MediaCodec.native_dequeueInputBuffer(Native Method)
E/ExoPlayerImplInternal(22389):       at android.media.MediaCodec.dequeueInputBuffer(MediaCodec.java:2726)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.mediacodec.SynchronousMediaCodecAdapter.dequeueInputBufferIndex(SynchronousMediaCodecAdapter.java:99)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.mediacodec.MediaCodecRenderer.feedInputBuffer(MediaCodecRenderer.java:1180)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:775)
E/ExoPlayerImplInternal(22389):       ... 5 more
I/HwExtendedUtils(22389): Set to window composer mode as 1
I/ACodec  (22389): gralloc usage: 0x80000000(OMX) => 0x80002900(ACodec)
D/SurfaceUtils(22389): disconnecting from surface 0x7450fea010, reason setNativeWindowSizeFormatAndUsage
E/ExoPlayerImplInternal(22389): Disable failed.
E/ExoPlayerImplInternal(22389):   java.lang.IllegalStateException
E/ExoPlayerImplInternal(22389):       at android.media.MediaCodec.native_flush(Native Method)
E/ExoPlayerImplInternal(22389):       at android.media.MediaCodec.flush(MediaCodec.java:2194)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.mediacodec.SynchronousMediaCodecAdapter.flush(SynchronousMediaCodecAdapter.java:166)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.mediacodec.MediaCodecRenderer.flushCodec(MediaCodecRenderer.java:858)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.mediacodec.MediaCodecRenderer.flushOrReleaseCodec(MediaCodecRenderer.java:851)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.mediacodec.MediaCodecRenderer.onDisabled(MediaCodecRenderer.java:691)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.video.MediaCodecVideoRenderer.onDisabled(MediaCodecVideoRenderer.java:609)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.BaseRenderer.disable(BaseRenderer.java:186)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.ExoPlayerImplInternal.disableRenderer(ExoPlayerImplInternal.java:1692)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.ExoPlayerImplInternal.resetInternal(ExoPlayerImplInternal.java:1426)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.ExoPlayerImplInternal.stopInternal(ExoPlayerImplInternal.java:1387)
E/ExoPlayerImplInternal(22389):       at com.bitmovin.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:604)
E/ExoPlayerImplInternal(22389):       at android.os.Handler.dispatchMessage(Handler.java:106)
E/ExoPlayerImplInternal(22389):       at android.os.Looper.loop(Looper.java:219)
E/ExoPlayerImplInternal(22389):       at android.os.HandlerThread.run(HandlerThread.java:67)
E/SurfaceUtils(22389): Failed to disconnect from surface 0x7450fea010, err -19
E/SurfaceUtils(22389): nativeWindowDisconnect failed: No such device (19)
E/ACodec  (22389): Failed to allocate output port buffers after port reconfiguration: (-19)
E/ACodec  (22389): signalError(omxError 0x80001001, internalError -19)
E/ACodec  (22389): Error occurred while disabling the output port
E/MediaCodec(22389): Codec reported err 0xffffffed, actionCode 0, while in state 0
I/M
```

File: android/src/main/kotlin/com/bitmovin/player/flutter/PlayerPlugin.kt [hunk 17]
```diff
@@ -54,14 +78,20 @@ class PlayerPlugin : FlutterPlugin, ActivityAware {
     }
 
     override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+        activity = WeakReference(binding.activity)
         flutterPluginBindingReference.get()?.let {
             register(it)
         }
+        if (BitmovinCastManager.isInitialized()) {
+            castManagerUpdateContext()
+        }
```
[the remark this answers is not in the document]
Comment (krocard) [reply android/src/main/kotlin/com/bitmovin/player/flutter/PlayerPlugin.kt:86]: Done, as discussed: 05965c5

File: example/lib/pages/cast.dart [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+  );
+
+  final Future<_PlayerState> _playerState = () async {
+    // The cast manager must be created before the player.
+    final castManager = await BitmovinCastManager.initialize();
+    final player = Player(
+      config: const PlayerConfig(
+        key: Env.bitmovinPlayerLicenseKey,
+        playbackConfig: PlaybackConfig(
+          isAutoplayEnabled: true,
```
[the remark this answers is not in the document]
Comment (krocard) [reply example/lib/pages/cast.dart:39]: Done

File: example/lib/main.dart [hunk 19]
```diff
@@ -33,6 +34,7 @@ class _MyAppState extends State<MyApp> {
         EventSubscription.routeName: (_) => const EventSubscription(),
         CustomHtmlUi.routeName: (_) => const CustomHtmlUi(),
         FullscreenHandling.routeName: (_) => const FullscreenHandling(),
+        Cast.routeName: (_) => const Cast(),
```
[the remark this answers is not in the document]
Comment (krocard) [reply example/lib/main.dart]: Done: bc3edabcc1ed913591c795e45be296d1205e1926

File: lib/src/casting_manager.dart
[the remark this answers is not in the document]
Comment (krocard) [reply lib/src/casting_manager.dart:1]: Done: 4af229059f9d1faec075d0a684d3b776b99e56e5

Comment (krocard) [reply lib/src/casting_manager.dart:1]: Done: 4af229059f9d1faec075d0a684d3b776b99e56e5

File: lib/src/player.dart (hunk 16 above)
Comment (krocard) [reply lib/src/player.dart]: Removed the async instead as it is not needed: 69c6b2ae80bdd5e50591c8acc7a8f8765d56e00c

File: lib/src/api/casting/casting_config.dart
[the remark this answers is not in the document]
Comment (krocard) [reply lib/src/api/casting/casting_config.dart:1]: Done: 376170fd02cab567e634eee5e52c89da2e225652

File: lib/src/api/player/player_api.dart (hunk 14 above)
Comment (krocard) [reply lib/src/api/player/player_api.dart]: Remove the second sentence: e008c24b9c2dfd8711042529c308760e210e478b

File: lib/src/api/player/player_api.dart (hunk 13 above)
Comment (krocard) [reply lib/src/api/player/player_api.dart:91]: Done: 8e5b06961ac4c8a380158b8f5712af8fa662f550

File: lib/src/casting/casting_config.dart
[the remark this answers is not in the document]
Comment (krocard) [reply lib/src/casting/casting_config.dart:1]: I was not aware of this convention. It should now be fixed: https://github.com/bitmovin/bitmovin-player-flutter/commit/376170fd02cab567e634eee5e52c89da2e225652

File: example/lib/pages/cast.dart [hunk 20]
```diff
@@ -0,0 +1,130 @@
...
+              child: const Text('Stop Casting'),
+            ),
+          ],
+        ),
+        Row(
+          children: [
+            OutlinedButton(
+              onPressed: castManager.updateContext,
+              child: const Text('Update cast context'),
+            ),
```
[the remark this answers is not in the document]
Comment (krocard) [reply example/lib/pages/cast.dart]: Removed as discussed.

File: lib/src/casting/casting_config.dart (hunk 2 above)
Comment (krocard) [reply lib/src/casting/casting_config.dart]: Done: 55631f45bc62942d63791d9ace89dc6afc2d4d16

File: lib/src/api/casting/casting_api.dart
[the remark this answers is not in the document]
Comment (krocard) [reply lib/src/api/casting/casting_api.dart:1]: Done

File: lib/src/api/casting/casting_config.dart (hunk 7 above)
Comment (krocard) [reply lib/src/api/casting/casting_config.dart]: Done

File: lib/bitmovin_player.dart
Comment (hawk23) [inline lib/bitmovin_player.dart:1]: I think `custom_cast_message.dart` is missing here

File: lib/src/api/player/player_config.dart [hunk 21]
```diff
@@ -1,4 +1,5 @@
 import 'package:bitmovin_player/bitmovin_player.dart';
+import 'package:bitmovin_player/src/api/casting/remote_control_config.dart';
```
Comment (hawk23) [inline lib/src/api/player/player_config.dart:2]: We can remove this import as it is included in 
```
import 'package:bitmovin_player/bitmovin_player.dart';
```

File: example/lib/pages/casting.dart [hunk 22]
```diff
@@ -0,0 +1,123 @@
...
+
+import 'package:bitmovin_player/bitmovin_player.dart';
+import 'package:bitmovin_player_example/controls.dart';
+import 'package:bitmovin_player_example/env/env.dart';
+import 'package:flutter/material.dart';
+
+class Casting extends StatefulWidget {
+  const Casting({super.key});
+
+  static String routeName = 'Cast';
```
Comment (hawk23) [inline example/lib/pages/casting.dart:11]: Super nit: 
```suggestion
  static String routeName = 'Casting';
```

**Review by hawk23 — APPROVED** [review]
Nice! Just some minor comments so approving already. The CI should also pass once those are addressed

File: lib/src/api/player/player_config.dart (hunk 21 above)
Comment (krocard) [reply lib/src/api/player/player_config.dart:2]:
```suggestion
```

File: lib/bitmovin_player.dart
Comment (hawk23) [reply lib/bitmovin_player.dart:1]: You are right. I missed that it is not used as an argument to `sendMessage`. My bad

**Pull request merged**